OPHD/Things/Structures/CommandCenter.h

```diff
@@ -1,14 +1,14 @@
 #pragma once
 
-#include "Structure.h"
+#include "FoodProduction.h"
 
 /**
  * Implements the Command Center structure.
  */
-class CommandCenter: public Structure
+class CommandCenter: public FoodProduction
```

Comment:
I see what you mean by counter-intuitive. I think this is partly why I have a sneaking suspicion that we might want a flatter structure hierarchy, and use a more data driven approach to control structure characteristics. Perhaps every structure can potentially have any feature, which is enabled/disabled by structure specific data. Think all structures having a power production value, but it being 0 for everything but power plants, and every structure having a food production capacity, but it being 0 for everything but agridomes. There are a lot of orthogonal concerns, which don't map well to inheritance (without introducing multiple inheritance), so a data driven approach might avoid some of the structure classification problems. Admitted though, I haven't thought deeply enough about the idea to know how well it would pan out.

Reply:
Agreed on many points though I hesitate to use a data driven approach for the simple fact that it winds up producing massive classes with a lot of flaggy behavior. It would reduce the need for casting downstream though so that would be a significant benefit.

The 'if' block was intended. I accidentally squashed too many commits into one when trying to clean up the history. It's a relatively new thing for me and a detail I missed.

To note it here, the 'if' block was actually wrong as the mFood variable is a cached value used for display purposes.